docker/freedv_build_windows.sh

```diff
@@ -1,22 +1,89 @@
-#!/bin/bash -x
+#!/bin/bash
 
-# Automation to cross compile freedv-gui for Windows using Docker
-#
-# usage:
-#   $ [FDV_GIT_REPO=your_repo] [FDV_GIT_BRANCH=your_branch] ./freedv_build_windows.sh 64|32
+function print_help {
+    echo
+    echo "Build freedv-gui for Windows using Docker"
+    echo
+    echo "  usage ./freedv_build_windows.sh [-d] [--noclean] [--build] [--repo GitRepo] [--branch GitBranch] 32|64"
+    echo
+    echo "    -d                  debug mode; trace script execution"
+    echo "    --noclean           start from a previous build (git pull && make), which is faster for small changes."
+    echo "                        The default is a clean build from a fresh git clone (slow but safer)"
+    echo "    --build             Rebuild docker image first (run if you have modifed the docker scripts)"
+    echo "    --repo GitRepo      (default https://github.com/drowe67/freedv-gui.git)"
+    echo "    --branch GitBranch  (default master)"
+    echo "    --bootstrap-wx      Builds wxWidgets from source (may take significantly longer to complete)"
+    echo
+    exit
+}
 
-[ -z $FDV_GIT_BRANCH ] && FDV_GIT_BRANCH=master
-[ -z $FDV_GIT_REPO ] && FDV_GIT_REPO=https://github.com/drowe67/freedv-gui.git
+# defaults - these variables are passed to the docker container
+FDV_CLEAN=1
+FDV_REBUILD=0
+FDV_GIT_REPO=https://github.com/drowe67/freedv-gui.git
+FDV_GIT_BRANCH=master
+FDV_BOOTSTRAP_WX=0
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]; do
+key="$1"
+case $key in
+    -d)
+        set -x	
+        shift
+    ;;
+    --noclean)
+        FDV_CLEAN=0	
+        shift
+    ;;
+    --build)
+        FDV_REBUILD=1
+        shift
+    ;;
+    --repo)
+        FDV_GIT_REPO="$2"	
+        shift
+        shift
+    ;;
+    --branch)
+        FDV_GIT_BRANCH="$2"	
+        shift
+        shift
+    ;;
+    --bootstrap-wx)
+        FDV_BOOTSTRAP_WX=1
+        shift
+    ;;
+    -h|--help)
+        print_help	
+    ;;
+    *)
+    POSITIONAL+=("$1") # save it in an array for later
+    shift
+    ;;
+esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
 
 FDV_CMAKE=mingw64-cmake
 if [ $# -eq 1 ]; then
     if [ $1 -eq 32 ]; then
         FDV_CMAKE=mingw32-cmake
     fi
+else
+    print_help
 fi
 
 log=build_log.txt
-FDV_CMAKE=$FDV_CMAKE FDV_GIT_REPO=$FDV_GIT_REPO FDV_GIT_BRANCH=$FDV_GIT_BRANCH docker-compose -f docker-compose-win.yml up > $log
+
+if [ $FDV_REBUILD -eq 1 ]; then
+    docker-compose -f docker-compose-win.yml rm -f > $log
+    docker-compose -f docker-compose-win.yml build --no-cache >> $log
+else
```

Comment:
@timw should split these steps up?  For example say we make a minor change to the docker script, should we have to rebuild the entire image (installing every rpm)?  I watching it run atm `tail -f buld_log.txt` and it's pretty slow.

Reply:
Hmm, guess I'm assuming there is a docker command to just install the latest docker script, while keeping everything else ...

Reply:
I think that's technically what build without --no-cache is supposed to do. On the other hand, I'm not sure if people will necessarily know that there might be new packages available; I thought it would have upgraded wxWidgets on its own, for instance.